### ISSUE TITLE
False Positive on Missing Comma

### ISSUE THREAD

**Issue by wyatt-herkamp** [rust-lang/rust-analyzer#17463]
**rust-analyzer version**: 
rust-analyzer version: 0.0.0 (e08f7953f 2024-06-20) [/media/Other/ProgrammingProjects/RustProjects/rust-analyzer/target/release/rust-analyzer]
I built this based on the latest commit today. 

**rustc version**: (eg. output of `rustc -V`)
rustc 1.79.0 (129f3b996 2024-06-10)

**editor or extension**: VSCode 1.90.1 OpenSuse Tumbleweed Linux 6.9.5

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)
[VS Code Settings](https://gist.github.com/wyatt-herkamp/23447c21b0550750be06ae080517cc0f)
**repository link (if public, optional)**: (eg. [rust-analyzer](https://github.com/rust-lang/rust-analyzer))
Private Repository however I was able to make a simple reproduction. 
**code snippet to reproduce**: 
```rust
use actix_web::{
    post,
    web::{self, Data},
    HttpResponse,
};
use sea_orm::{prelude::*, ActiveValue, DatabaseConnection};

fn main() {
    println!("Hello, world!");
}

use serde::{Deserialize, Serialize};
#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "users")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i64,
    pub name: String,
    pub username: String,
    pub email: String,
    #[serde(skip_serializing)]
    pub password: String,
    pub admin: bool,
    #[sea_orm(default_expr = "Expr::current_timestamp()")]
    pub created: DateTimeWithTimeZone,
}
impl ActiveModelBehavior for ActiveModel {}
#[derive(Deserialize)]
pub struct NewUser {
    pub username: String,
    pub password: String,
    pub email: String,
    pub name: String,
}
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

#[post("/install")]
pub async fn install(
    post: web::Json<NewUser>,
    database: Data<DatabaseConnection>,
) -> Result<HttpResponse, actix_web::error::Error> {
    let post = post.into_inner();
    let user = ActiveModel {
        username: ActiveValue::Set(post.username),
        password: ActiveValue::Set(post.password),
        email: ActiveValue::Set(post.email),
        name: ActiveValue::Set(post.name),
        admin: ActiveValue::Set(true),
        ..Default::default()
    };
    Entity::insert(user).exec(database.as_ref()).await.unwrap();

    Ok(HttpResponse::Ok().finish())
}
```
![image](https://github.com/rust-lang/rust-analyzer/assets/11785959/b84d56b5-6c20-4dfe-8153-6996c55b06d7)

### So when I expand the post macro I get this. 
![image](https://github.com/rust-lang/rust-analyzer/assets/11785959/db03fc81-283d-4282-9935-832330e684b5)

I was going to try to fix it however, I have no idea where it is coming from. If someone has an idea where it is coming from let me know I will try to fix it myself. 

If I move to the stable version of rust-analyzer so the release on 17th I have no issues. But the nightly release has issues.

**Comment by wyatt-herkamp** [comment]
https://github.com/rust-lang/rust-analyzer/commit/158626bed4a34e80d431acb74bc352364e2cebc7

I think it is this commit

**Comment by Veykril** [comment]
Right, https://github.com/rust-lang/rust-analyzer/commit/158626bed4a34e80d431acb74bc352364e2cebc7#diff-f002869811628add2076bd19a6d7b800a0186e9fcc946e953afa8c981ce342b0R278-R289 is wrong, that also needs to check that the expression is missing

**Comment by Veykril** [comment]
That is this test will currently result in that wrong expect when it shouldnt trigger there at all
```rs

    #[test]
    fn fixup_record_ctor_field() {
        check(
            r#"
fn foo() {
    R { f: a }
}
"#,
            expect![[r#"
fn foo () {R {f : a __ra_fixup}}
"#]],
        )
    }
```

**Comment by wyatt-herkamp** [comment]
> Right, [158626b#diff-f002869811628add2076bd19a6d7b800a0186e9fcc946e953afa8c981ce342b0R278-R289](https://github.com/rust-lang/rust-analyzer/commit/158626bed4a34e80d431acb74bc352364e2cebc7#diff-f002869811628add2076bd19a6d7b800a0186e9fcc946e953afa8c981ce342b0R278-R289) is wrong, that also needs to check that the expression is missing

LOL, I was literally typing I figured out what the problem code was.  So just check that `it.expr().is_none()`

**Comment by wyatt-herkamp** [comment]
> That is this test will currently result in that wrong expect when it shouldnt trigger there at all
> 
> ```rust
>     #[test]
>     fn fixup_record_ctor_field() {
>         check(
>             r#"
> fn foo() {
>     R { f: a }
> }
> "#,
>             expect![[r#"
> fn foo () {R {f : a __ra_fixup}}
> "#]],
>         )
>     }
> ```


I added the `it.expr().is_none()` and the tests past and my code runs fine.
